packages/edit-widgets/src/store/actions.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	parse,
+	serialize,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import {
+	select,
+	apiFetch,
+} from './controls';
+
+export function setupWidgetAreas( widgetAreas ) {
+	return {
+		type: 'SETUP_WIDGET_AREAS',
+		widgetAreas: map( widgetAreas, ( area ) => {
+			return {
+				...area,
+				blocks: parse( ( area.content && area.content.raw ) || '' ),
+			};
+		} ),
+	};
+}
+
+export function updateBlocksInWidgetArea( widgetAreaId, blocks ) {
+	return {
+		type: 'UPDATE_BLOCKS_IN_WIDGET_AREA',
+		widgetAreaId,
+		blocks,
+	};
+}
+
+export function* saveWidgetAreas() {
+	const widgetAreas = yield select(
+		STORE_KEY,
+		'getWidgetAreas'
+	);
+	for ( const widgetArea of widgetAreas ) {
```

Comment:
This makes me wonder if we needed a "batched" save endpoint. I know there's no precedent in the REST API endpoints about this though.

Reply:
I guess most of the times the user will not change all the areas, and soon as we have some kind of "isDirty" dection we will only save the changed areas.

Reply:
I was contemplating having a `POST` endpoint which can update multiple areas while writing https://github.com/WordPress/gutenberg/pull/14812. Probably best to not get ahead of ourselves, though, as the design may yet change.